web/plugs/JWTverifier.ex

```diff
@@ -0,0 +1,59 @@
+require IEx
+defmodule Janitor.Plugs.JWTVerifier do
+  use Timex
+  import Plug.Conn
+
+  def init(_) do
+  end
+
+  def call(conn, _) do
+    case verification_chain(conn) do
+      {:ok, conn} -> conn
+      {:error, message} -> send_resp(conn, 403, message) |> Plug.Conn.halt
+    end
+  end
+
+  #PRIVATE
+
+  defp verification_chain(conn) do
+    with {:ok, conn} <- verify_token(conn),
```

Comment:
@mikolajgrzyb No, because rescue block does not halt the pipeline - it rescues the error, after all :)